src/ngraph/runtime/cpu/cpu_external_function.cpp

```diff
@@ -1353,6 +1351,65 @@ void runtime::cpu::CPU_ExternalFunction::build()
         enable_nodename_list.emplace_back(make_pair(enable, node->get_name()));
     }
 
+    if ((std::getenv("NGRAPH_DEX_DEBUG") != nullptr))
+    {
+        string filename = file_util::path_join(s_debug_dir, m_function_name + "_debug.txt");
+        std::stringstream strm;
+        auto find_role = [&](CPUTensorRole tensor_role) -> string {
+            switch (tensor_role)
+            {
+            case CPUTensorRole::INPUT: return string("CPUTensorRole::INPUT");
+            case CPUTensorRole::INTERMEDIATE: return string("CPUTensorRole::INTERMEDIATE");
+            case CPUTensorRole::CONSTANT: return string("CPUTensorRole::CONSTANT");
+            case CPUTensorRole::OUTPUT: return string("CPUTensorRole::OUTPUT");
+            }
+        };
+
+        //dump the tensor roles to debug manifest
+        for (const auto& tensor_roles : m_tensor_roles)
+        {
+            strm << tensor_roles.first << ", " << find_role(tensor_roles.second) << "\n";
+        }
+
+        write_to_file(strm.str(), s_debug_dir, filename);
+        strm.str("");
+
+        //dump the op's the order of execution along with their memory references
+        for (shared_ptr<Node> node : m_function->get_ordered_ops())
+        {
+            std::vector<string> node_inputs;
+            std::vector<string> node_outputs;
+            std::stringstream temp;
+            if (node->is_parameter() || node->is_constant())
+            {
+                continue;
+            }
+            for (const descriptor::Input& input : node->get_inputs())
+            {
+                const descriptor::Output& output = input.get_output();
+                shared_ptr<descriptor::TensorView> tv = output.get_tensor_ptr();
+                temp << &tensor_data[tv->get_name()];
```

Comment:
I don't think &tensor_data[tv->get_name()] gets you the tensor base address but not sure what the intent was here. At any rate, that information isn't even available at compile time. 
Could you add a very brief note explaining why the address of tensor_data mapped values is useful?

Reply:
sure, I can add the note. Yes, its true we can't get the tensor base address during compile time but we can still refer to the address of the tensor_data(which will be unique to each tensor) which holds the base address. The idea is to use this uniq address in debug manifest, but their is one level indirection thou. 

My original implementation is to actually dump the base address but @jbobba suggested this way to simply the executor code.

Reply:
👍